test/e2e/space_autocompletion_test.go

```diff
@@ -0,0 +1,157 @@
+package e2e
+
+import (
+	"testing"
+	"time"
+
+	toolchainv1alpha1 "github.com/codeready-toolchain/api/api/v1alpha1"
+	testconfig "github.com/codeready-toolchain/toolchain-common/pkg/test/config"
+	. "github.com/codeready-toolchain/toolchain-e2e/testsupport"
+	"github.com/codeready-toolchain/toolchain-e2e/testsupport/wait"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestAutomaticClusterAssignment(t *testing.T) {
+	// given
+	awaitilities := WaitForDeployments(t)
+	hostAwait := awaitilities.Host()
+	memberAwait1 := awaitilities.Member1()
+	memberAwait2 := awaitilities.Member2()
+	// TODO: we need to create some users to be able to limit the capacity of the clusters. The code won't be needed as soon as we start counting Spaces instead of UserAccounts https://issues.redhat.com/browse/CRT-1427
+	NewSignupRequest(t, awaitilities).
+		Username("for-member1").
+		Email("for-member1@redhat.com").
+		TargetCluster(memberAwait1).
+		ManuallyApprove().
+		RequireConditions(ConditionSet(Default(), ApprovedByAdmin())...).
+		EnsureMUR().
+		Execute()
+	NewSignupRequest(t, awaitilities).
+		Username("for-member2").
+		Email("for-member2@redhat.com").
+		TargetCluster(memberAwait2).
+		RequireConditions(ConditionSet(Default(), ApprovedByAdmin())...).
+		ManuallyApprove().
+		EnsureMUR().
+		Execute()
+	hostAwait.UpdateToolchainConfig(testconfig.Tiers().DefaultSpaceTier("appstudio"))
+
+	t.Run("set low capacity threshold and expect that space will have default tier, but won't have target cluster so it won't be provisioned", func(t *testing.T) {
```

Comment:
that is a _long_ test name! 😂

Reply:
it's self-explanatory :stuck_out_tongue_closed_eyes: :smile:

Reply:
definitely! 😂